### DISCUSSION TITLE
Route for listing all available model types

### DISCUSSION THREAD

**Discussion by ImDarkTom** [comfyanonymous/ComfyUI#4833] [Ideas]
Can we add a route that allows listing all available extra model paths/model types?

**Justification**:

I'm currently developing a project that relies heavily on the comfyui api and from what I've found there is no reliable way to check what extra search paths have been added using a route. 

There is `/internal/files?directory=models` however documentation says that internal routes should not be relied upon as they may be subject to change.

Could we perhaps see a route added that allows for this sort of thing?

**Comment by ltdrdata** [comment]
You can easily develop an extension that implements a custom endpoint.

https://github.com/ltdrdata/ComfyUI-Inspire-Pack/blob/25ae8d522f781f694bf736f87945f2291da531b4/inspire/inspire_server.py#L14

**Reply by schuster-rainer** [reply]
but that's the point, not having to do that manually but have a reliable interface in core to list the configured directories
